test/integration/sdf_custom.cc

```diff
@@ -82,3 +82,222 @@ TEST(SDFParser, CustomElements)
       tranJointElement->Get<std::string>("mysim:hardwareInterface");
   EXPECT_EQ("EffortJointInterface", tranHwInterface);
 }
+
+/////////////////////////////////////////////////
+TEST(SDFParser, ReloadCustomElements)
+{
+  const std::string sdfTestFile =
+      sdf::testing::TestFile("integration", "custom_elems_attrs.sdf");
+
+  // load file with custom elements
+  sdf::Root root1;
+  sdf::Errors errors = root1.Load(sdfTestFile);
+  EXPECT_TRUE(errors.empty());
+
+  // reload string output of root1
+  sdf::Root root2;
+  errors = root2.LoadSdfString(root1.Element()->ToString(""));
+  EXPECT_TRUE(errors.empty());
+
+  // check that root1 and root2 equal
+  const sdf::World *world1 = root1.WorldByIndex(0);
+  const sdf::World *world2 = root2.WorldByIndex(0);
+  ASSERT_NE(nullptr, world1);
+  ASSERT_NE(nullptr, world2);
+
+  const sdf::Model *model1 = world1->ModelByIndex(0);
+  const sdf::Model *model2 = world2->ModelByIndex(0);
+  ASSERT_NE(nullptr, model1);
+  ASSERT_NE(nullptr, model2);
+  EXPECT_EQ(model1->Element()->ToString(""), model2->Element()->ToString(""));
+
+  const sdf::Link *link1 = model1->LinkByIndex(0);
+  const sdf::Link *link2 = model2->LinkByIndex(0);
+  ASSERT_NE(nullptr, link1);
+  ASSERT_NE(nullptr, link2);
+  EXPECT_EQ(link1->Element()->ToString(""), link2->Element()->ToString(""));
+
+  sdf::ElementPtr customElem1 =
+      model1->Element()->FindElement("mysim:transmission");
+  sdf::ElementPtr customElem2 =
+      model2->Element()->FindElement("mysim:transmission");
+  ASSERT_NE(nullptr, customElem1);
+  ASSERT_NE(nullptr, customElem2);
+  EXPECT_EQ(customElem1->ToString(""), customElem2->ToString(""));
+  sdf::ElementPtr customDesc1 =
+      world1->Element()->FindElement("mysim:description");
+  sdf::ElementPtr customDesc2 =
+      world2->Element()->FindElement("mysim:description");
+  ASSERT_NE(nullptr, customDesc1);
+  ASSERT_NE(nullptr, customDesc2);
+  EXPECT_EQ(customDesc1->ToString(""), customDesc2->ToString(""));
```

Comment:
Can we compare the result of `customDesc2->ToString` with the expected string literal instead of trusting `customDesc1->ToString` is working properly? Here and below. Not to make to verbose, I would only do it on the comparisons that involve custom elements.

Reply:
c54dccb0f328c0f7e1f00339ed0d740f963faddd